File: src/contexts/core/kanji/application/find-one/find-kanji-query.ts
import type { Query } from "../../../../shared/cqrs/domain/query";

class FindKanjiQuery implements Query {
    
        static readonly TYPE = "tanukanji.core.kanji.find:1.0.0";
    
        constructor(
            public readonly ideogram: string
        ) {}
    
        get type(): string {
            return FindKanjiQuery.TYPE;
        }
}

export { FindKanjiQuery };

File: libs/kernel/src/kanjis/application/find_one/find_kanji_query.rs
use std::any::Any;

use cqrs::domain::query::Query;


pub struct FindKanjiQuery {
    pub ideogram: String,
}

impl FindKanjiQuery {

    pub const QUERY_TYPE: &'static str = "FindKanjiQuery";

    pub fn new(ideogram: String) -> Self {
        Self { ideogram }
    }
}

impl Query for FindKanjiQuery {
    fn get_type(&self) -> String {
        Self::QUERY_TYPE.to_string()
    }

    fn as_any(&self) -> &dyn Any {
        self
    }
}